### ISSUE TITLE
Cannot make z88dk

### ISSUE THREAD

**Issue by SM7I** [spectrumero/spectranet#11]
After git pull today I can no longer make z88dk

Usual make works fine, but make z88dk gives me:

Error: object file 'addFormData.o' version 12, expected version 14
Error: object file 'freeFormData.o' version 12, expected version 14
Error: object file 'parseURI.o' version 12, expected version 14
Error: object file 'parseproto.o' version 12, expected version 14
Error: object file 'allocURI.o' version 12, expected version 14
Error: object file 'freeURI.o' version 12, expected version 14
Error: object file 'base64enc.o' version 12, expected version 14
Error: object file 'request.o' version 12, expected version 14
Error: object file 'readData.o' version 12, expected version 14
Error: object file 'postsize.o' version 12, expected version 14
Error: object file 'getheader.o' version 12, expected version 14
Error: object file 'freeheaders.o' version 12, expected version 14
Error: object file 'readHeaders.o' version 12, expected version 14
Error: object file 'htrecv.o' version 12, expected version 14

What to do ?

**Comment by ZXGuesser** [comment]
That means you're trying to link version 12 object files with version 14.
`make clean` and try again.

**Comment by SM7I** [comment]
Unfortunately that didn´t help.....

**Comment by SM7I** [comment]
Now I´ve deleted my git and done a new clone. Now I get further, but I now get error for the switch -preserve to zcc. Doesn´t seem to exist...

**Comment by feilipu** [comment]
Try removing all the binaries in the ./bin directory. That will trigger them to be rebuilt, and might help.

**Comment by SM7I** [comment]
Did that and also make clean, but not improving. Still same problem....

**Comment by thweasel** [comment]
Having updated to Z88DK 2.1 (release version), building the z88dk spectranet libraries I have found the same issue with -preserve in the Makefile in z88dk/libhttp. I am not using libhttp so no idea if there are side effects.

The other annoying little issue was with the link z88dk/libspectranet/spectranet.asm -> z88dk/socklib/spectranet.asm was broken and was reporting the file "missing", which was a little misleading.

**Comment by desertkun** [comment]
I've had these, too. The solution is to remove that flag and to copy that file. Maybe you can create a PR to fix this?

**Comment by ZXGuesser** [comment]
Yes that weird broken symlink goes right back to the old subversion repo. I guess svn does some magic symlink translation on unix.

**Comment by ZXGuesser** [comment]
#18 should cure both issues
